TestLimitRanger_GetLimitRangesFixed22422 flakes

### Which jobs are flaking?

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/111095/pull-kubernetes-unit/1585051220769247232
```
panic: test timed out after 3m0s

goroutine 112 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:2036 +0xbb
created by time.goFunc
	/usr/local/go/src/time/sleep.go:176 +0x48
```

### Which tests are flaking?

pull-kubernetes-unit

### Since when has it been flaking?

https://github.com/kubernetes/kubernetes/pull/112696


### Testgrid link

https://storage.googleapis.com/k8s-triage/index.html?test=TestLimitRanger_GetLimitRangesFixed22422#658be1b0e17b463179af

### Reason for failure (if possible)

```
goroutine 89 [chan receive, 2 minutes]:
k8s.io/kubernetes/plugin/pkg/admission/limitranger.TestLimitRanger_GetLimitRangesFixed22422.func1({0x2bf1100, 0xc0006ea240})
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/plugin/pkg/admission/limitranger/admission_test.go:885 +0x19e
k8s.io/kubernetes/vendor/k8s.io/client-go/testing.(*SimpleReactor).React(0xc0006b1d10, {0x2bf1100, 0xc0006ea240})
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/testing/fixture.go:530 +0x5d
k8s.io/kubernetes/vendor/k8s.io/client-go/testing.(*Fake).Invokes(0xc0006ca000, {0x2bf1100, 0xc00053c480}, {0x2bdaa98, 0xc000781618})
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/testing/fake.go:145 +0x389
k8s.io/kubernetes/vendor/k8s.io/client-go/kubernetes/typed/core/v1/fake.(*FakeLimitRanges).List(0xc0000122d0, {0xc00070c0f0?, 0x0?}, {{{0x0, 0x0}, {0x0, 0x0}}, {0x0, 0x0}, {0x0, ...}, ...})
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/client-go/kubernetes/typed/core/v1/fake/fake_limitrange.go:60 +0x265
k8s.io/kubernetes/plugin/pkg/admission/limitranger.(*LimitRanger).GetLimitRanges.func1()
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/plugin/pkg/admission/limitranger/admission.go:169 +0x14a
k8s.io/kubernetes/vendor/golang.org/x/sync/singleflight.(*Group).doCall.func2(0xc000781b67, 0xc00044c480, 0xc0004a2d98)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/golang.org/x/sync/singleflight/singleflight.go:193 +0xb5
k8s.io/kubernetes/vendor/golang.org/x/sync/singleflight.(*Group).doCall(0xc000298040, 0xc00044c480, {0x2695e14, 0x4}, 0x238d420?)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/golang.org/x/sync/singleflight/singleflight.go:195 +0x112
k8s.io/kubernetes/vendor/golang.org/x/sync/singleflight.(*Group).Do(0xc000298040, {0x2695e14, 0x4}, 0x0?)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/golang.org/x/sync/singleflight/singleflight.go:108 +0x211
k8s.io/kubernetes/plugin/pkg/admission/limitranger.(*LimitRanger).GetLimitRanges(0xc000298000, {0x2bf6cf0, 0xc000304d80})
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/plugin/pkg/admission/limitranger/admission.go:168 +0x2cf
k8s.io/kubernetes/plugin/pkg/admission/limitranger.TestLimitRanger_GetLimitRangesFixed22422.func2()
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/plugin/pkg/admission/limitranger/admission_test.go:904 +0xd0
created by k8s.io/kubernetes/plugin/pkg/admission/limitranger.TestLimitRanger_GetLimitRangesFixed22422
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/plugin/pkg/admission/limitranger/admission_test.go:902 +0x16a9

```

### Anything else we need to know?

/cc @aimuz  @aojea 

### Relevant SIG(s)


/sig scalability

--- a/plugin/pkg/admission/limitranger/admission_test.go
+++ b/plugin/pkg/admission/limitranger/admission_test.go
@@ -864,12 +864,21 @@ func TestLimitRanger_GetLimitRangesFixed22422(t *testing.T) {
 	limitRange := validLimitRangeNoDefaults()
 	limitRanges := []corev1.LimitRange{limitRange}
 
-	var count int64
 	mockClient := &fake.Clientset{}
 
-	unhold := make(chan struct{})
+	var (
+		testCount  int64
+		test1Count int64
+	)
 	mockClient.AddReactor("list", "limitranges", func(action core.Action) (bool, runtime.Object, error) {
-		atomic.AddInt64(&count, 1)
+		switch action.GetNamespace() {
+		case "test":
+			atomic.AddInt64(&testCount, 1)
+		case "test1":
+			atomic.AddInt64(&test1Count, 1)
+		default:
+			t.Error("unexpected namespace")
+		}
 
 		limitRangeList := &corev1.LimitRangeList{
 			ListMeta: metav1.ListMeta{
@@ -881,8 +890,8 @@ func TestLimitRanger_GetLimitRangesFixed22422(t *testing.T) {
 			value.Namespace = action.GetNamespace()
 			limitRangeList.Items = append(limitRangeList.Items, value)
 		}
-		// he always blocking before sending the signal
-		<-unhold
+		// make the handler slow so concurrent calls exercise the singleflight
+		time.Sleep(time.Second)
 		return true, limitRangeList, nil
 	})
 
@@ -926,34 +935,30 @@ func TestLimitRanger_GetLimitRangesFixed22422(t *testing.T) {
 			}
 		}()
 	}
-	// unhold all the calls with the same namespace handler.GetLimitRanges(attributes) calls, that have to be aggregated
-	unhold <- struct{}{}
-	go func() {
-		unhold <- struct{}{}
-	}()
 
 	// and here we wait for all the goroutines
 	wg.Wait()
 	// since all the calls with the same namespace will be holded, they must be catched on the singleflight group,
 	// There are two different sets of namespace calls
 	// hence only 2
-	if count != 2 {
-		t.Errorf("Expected 1 limit range, got %d", count)
+	if testCount != 1 {
+		t.Errorf("Expected 1 limit range call, got %d", testCount)
+	}
+	if test1Count != 1 {
+		t.Errorf("Expected 1 limit range call, got %d", test1Count)
 	}
 
 	// invalidate the cache
 	handler.liveLookupCache.Remove(attributes.GetNamespace())
-	go func() {
-		// unhold it is blocking until GetLimitRanges is executed
-		unhold <- struct{}{}
-	}()
 	_, err = handler.GetLimitRanges(attributes)
 	if err != nil {
 		t.Errorf("unexpected error: %v", err)
 	}
-	close(unhold)
 
-	if count != 3 {
-		t.Errorf("Expected 2 limit range, got %d", count)
+	if testCount != 2 {
+		t.Errorf("Expected 2 limit range call, got %d", testCount)
+	}
+	if test1Count != 1 {
+		t.Errorf("Expected 1 limit range call, got %d", test1Count)
 	}
 }

